honeybee_vtk/scene.py

```diff
@@ -117,71 +196,34 @@ def get_legend(self, color_range) -> vtk.vtkScalarBarWidget():
 
         # create the scalar_bar_widget
         scalar_bar_widget = vtk.vtkScalarBarWidget()
-        scalar_bar_widget.SetInteractor(self._interactor)
+        scalar_bar_widget.SetInteractor(interactor)
         scalar_bar_widget.SetScalarBarActor(scalar_bar)
         return scalar_bar_widget
 
-    def add_model(self, model: Model):
-        """Add a model to scene."""
-        for ds in model:
-            if ds.is_empty:
-                continue
-            self.add_dataset(ds)
-
-    def add_dataset(self, data_set: ModelDataSet):
-        """Create a dataset to scene as a VTK actor."""
-
-        # calculate point data based on cell data
-        cell_to_point = vtk.vtkCellDataToPointData()
-        if len(data_set.data) > 1:
-            polydata = JoinedPolyData.from_polydata(data_set.data)
-            cell_to_point.SetInputConnection(polydata.GetOutputPort())
-        else:
-            polydata = data_set.data[0]
-            cell_to_point.SetInputData(polydata)
-
-        mapper = vtk.vtkPolyDataMapper()
-        mapper.SetInputConnection(cell_to_point.GetOutputPort())
-
-        # map cell data to pointdata
-        if data_set.fields_info:
-            field_info = data_set.active_field_info
-            mapper.SetColorModeToMapScalars()
-            mapper.SetScalarModeToUsePointData()
-            mapper.SetScalarVisibility(True)
-            range_min, range_max = field_info.data_range
-            mapper.SetScalarRange(range_min, range_max)
-            mapper.SetLookupTable(field_info.color_range())
-            mapper.Update()
-
-        actor = vtk.vtkActor()
-        actor.SetMapper(mapper)
-
-        if data_set.edge_visibility:
-            actor.GetProperty().EdgeVisibilityOn()
-
-        if data_set.display_mode == DisplayMode.Wireframe:
-            actor.GetProperty().SetRepresentationToWireframe()
-
-        self._renderer.AddActor(actor)
-
-    def to_gltf(self, folder, name):
-        """Save the scene to a glTF file.
+    def export_gltf(self, folder, renderer, window, name):
+        """Export a scene to a glTF file.
 
         Args:
             folder: A valid path to where you'd like to write the gltf file.
+            renderer: A vtk renderer object created using create_render_window method.
```

Comment:
Well, it can help if we create class for window. The same thing is happening at export_image method where we're using a window and an interactor object.

Reply:
As long as there is a default window/renderer - I think the idea of having a class for window to make it easier for one to create a rendered makes a lot of sense.

Reply:
Revised. A new module is created.